src/main/java/com/nvidia/spark/rapids/jni/iceberg/IcebergDateTimeUtil.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.jni.iceberg;
+
+import ai.rapids.cudf.*;
+
+/**
+ * Gpu implementations for `org.apache.iceberg.util.DateTimeUtil` functions
+ */
+public class IcebergDateTimeUtil {
+  static {
+    NativeDepsLoader.loadNativeDeps();
+  }
+
+  private static void checkTimestampOrDateType(ColumnView input) {
+    DType type = input.getType();
+    if (type.getTypeId() != DType.DTypeEnum.TIMESTAMP_MICROSECONDS &&
+        type.getTypeId() != DType.DTypeEnum.TIMESTAMP_DAYS) {
+      throw new IllegalArgumentException("Input column must be of type " +
+          "TIMESTAMP_MICROSECONDS or TIMESTAMP_DAYS");
+    }
+  }
+
+  private static void checkTimestampType(ColumnView input) {
+    DType type = input.getType();
+    if (type.getTypeId() != DType.DTypeEnum.TIMESTAMP_MICROSECONDS) {
+      throw new IllegalArgumentException("Input column must be of type " +
+          "TIMESTAMP_MICROSECONDS");
+    }
+  }
+
+  /**
+   * Calculates the difference in years between the epoch year (1970) and the
+   * given date/timestamp column. E.g.: for date '1971-01-01', the result would be
+   * 1: (1 year after epoch year)
+   *
+   * @param input The input date/timestamp column.
+   * @return A column of type INT32 containing the year differences from epoch.
+   */
+  public static ColumnVector toYears(ColumnView input) {
```

Comment:
Since this is a difference operation, it would be good to reflect that in the name.  Say:

```suggestion
  public static ColumnVector yearsFromEpoch(ColumnView input) {
```

Also, it might be obvious, but there might be value in mentioning that we expect negative values when the input date precedes epoch.